### PR TITLE
Update permissions for "CiviCRM Admin Menu -> Mailings -> Message Templates" nav item

### PR DESCRIPTION
**Pull request by alifrumin**
Overview
----------------------------------------
This builds on https://github.com/civicrm/civicrm-core/pull/11974. It makes the permissions needed to view the "Message Templates" menu item in the "Mailings" dropdown match the permissions needed to view the "Message Templates" page generally. 

Before
----------------------------------------
To view the menu item "CiviCRM Admin Menu -> Mailings -> Message Templates" the user has to have the permission 'edit message templates' this means if the user has the permission "edit user-driven message templates" and/or "edit system workflow message templates" but not "edit message templates" they cannot see this link even though they do have access to the page.

After
----------------------------------------
To view the menu item "CiviCRM Admin Menu -> Mailings -> Message Templates" the user has to have one of the following permissions: 
- edit user-driven message templates 
- edit system workflow message templates 
- edit message templates

### CODE REVIEW

File: CRM/Upgrade/Incremental/sql/5.34.alpha1.mysql.tpl [hunk 1]
```diff
@@ -9,3 +9,10 @@ INSERT IGNORE INTO `civicrm_state_province` (`id`, `country_id`, `abbreviation`,
 DELETE FROM civicrm_mapping_field
 WHERE name NOT IN ( SELECT concat('custom_', id) FROM civicrm_custom_field)
 AND name LIKE 'custom_%';
+
+-- Update permissions for CiviCRM Admin Menu -> CiviMail -> Message Templates
+UPDATE civicrm_navigation n
+SET n.permission = 'edit user-driven message templates,edit message templates,edit system workflow message templates'
+AND n.permission_operator = 'OR'
+WHERE n.name = 'Message Templates'
+AND n.permission = 'edit message templates'
```
Comment (seamuslee001) [inline CRM/Upgrade/Incremental/sql/5.34.alpha1.mysql.tpl:18]: @alifrumin missing a `;` here

Comment (alifrumin) [reply CRM/Upgrade/Incremental/sql/5.34.alpha1.mysql.tpl]: THANKS!!!

**Comment by seamuslee001** [comment]
@alifrumin upgrade issue

```
 * ERROR DEBUGINFO: UPDATE civicrm_navigation n
SET n.permission = 'edit user-driven message templates,edit message templates,edit system workflow message templates'
AND n.permission_operator = 'OR'
WHERE n.name = 'Message Templates'
AND n.permission = 'edit message templates' [nativecode=1292 ** Truncated incorrect DOUBLE value: 'edit user-driven message templates,edit message templates,edit system workflow message templates']
```

**Comment by alifrumin** [comment]
Thanks Seamus, I should have tagged this as WIP, its my first attempt and an upgrader script in awhile and I wanted to see jenkins test it.

**Comment by alifrumin** [comment]
I tested this on the jenkins build and it worked as expected and is now passing all tests. This is ready for review.

File: xml/templates/civicrm_navigation.tpl [hunk 2]
```diff
@@ -183,7 +183,7 @@ VALUES
...
     ( @domainID, 'civicrm/admin/component?reset=1',                         '{ts escape="sql" skip="true"}Headers, Footers, and Automated Messages{/ts}', 'Headers, Footers, and Automated Messages', 'access CiviMail,administer CiviCRM', 'AND', @mailinglastID, '1', NULL, 6 ),
-    ( @domainID, 'civicrm/admin/messageTemplates?reset=1',                  '{ts escape="sql" skip="true"}Message Templates{/ts}', 'Message Templates',                 'edit message templates', '', @mailinglastID, '1', NULL, 7 ),
+    ( @domainID, 'civicrm/admin/messageTemplates?reset=1',                  '{ts escape="sql" skip="true"}Message Templates{/ts}', 'Message Templates',                 'edit message templates,edit user-driven message templates,edit system workflow message templates', 'OR', @mailinglastID, '1', NULL, 7 ),
     ( @domainID, 'civicrm/admin/options/from_email_address?reset=1', '{ts escape="sql" skip="true"}From Email Addresses{/ts}', 'From Email Addresses', 'administer CiviCRM', '', @mailinglastID, '1', 1, 8 ),
```
Comment (demeritcowboy) [inline xml/templates/civicrm_navigation.tpl:187]: Note that there's 2 of these. There's also one under Admin - Communications.

Comment (demeritcowboy) [reply xml/templates/civicrm_navigation.tpl:187]: Actually there's 3. Also under admin - civimail.

Comment (alifrumin) [reply xml/templates/civicrm_navigation.tpl:187]: Yes, for which the permission is "Administer CiviCRM". 

In the scenario I am focused on the user does not have the permission "Administer CiviCRM" so they cannot see the "Administer" menu at all. But they can see the "Mailings" menu.

**Comment by demeritcowboy** [comment]
Ok gotcha. I might follow up after with a code comment in navigation.tpl explaining why they're different. So just civicrm_generated needs updating but otherwise looks good.

* General standards
    * [`r-explain`] __PASS__
    * [`r-user`] __PASS__
      * It's not from this PR but there seems to be some kind of glitch where the menu edit UI shows two "edit message templates" choices and the second one has the description for a dedupe permission. TBD.
    * [`r-doc`] __PASS__
    * [`r-run`] __PASS__
      * Upgrade works.
* Developer standards
    * [`r-tech`] __PASS__
    * [`r-code`] __Issue__
      * civicrm_generated.mysql needs updating.
    * [`r-maint`] ____
    * [`r-test`] __PASS__

**Comment by demeritcowboy** [comment]
Have added https://lab.civicrm.org/dev/core/-/issues/2291 for the selection glitch.

**Comment by eileenmcnaughton** [comment]
Yay thank you @demeritcowboy  - I was looking at this & thinking 'this looks pretty straight forward but argh I don't have my head into it' and then I got to the bottom & saw you had done a proper review! 

Merging

**Comment by eileenmcnaughton** [comment]
Just noting we do need the sql regeneration if someone can put that up

**Comment by demeritcowboy** [comment]
@seamuslee001 the upgrade script for this is in 5.34.alpha1 but this won't be in the 5.34 tree. How to fix? Revert script and redo for master?

**Comment by alifrumin** [comment]
@demeritcowboy thank you for your help on this!!!! When we resolve the tree issue I am happy to make a new pr that has the upgrade script in the proper file, additionally I am happy to attempt to run GenCode... as I understand it the documentation for that is here: https://docs.civicrm.org/dev/en/latest/framework/database/schema-definition/ 
> To update any DAO file you need to update the relevant XML and then run php Gencode.php from the XML directory (civibuild create will also do this, albeit rebuilding your whole dev site)

**Comment by demeritcowboy** [comment]
Hi, 19363 and 19364 are already done by @seamuslee001 . Gencode unfortunately doesn't generate civicrm_generated. For that file you either need to [run regen.sh](https://docs.civicrm.org/dev/en/latest/framework/database/#binregensh) or if it doesn't change ids you can get away with editing directly.

**Comment by alifrumin** [comment]
Alright!! beat me to it. I can try running regen.sh later this week.

**Comment by demeritcowboy** [comment]
You don't have to, unless you want the practice or the fun of accidentally wiping the wrong database. It's already done in 19363.

**Comment by alifrumin** [comment]
Well that’s just phenomenal. Thank you!!!

**Pull request merged**